src/datepicker/calendar.js

```diff
@@ -552,7 +552,9 @@ export default class Calendar extends React.Component<
           <Root
             data-baseweb="calendar"
             ref={root => {
-              this.root = root;
+              if (!root && this.state.rootElement) {
```

Comment:
Why `!root`?

Reply:
Yea sorry, this was incorrect and it was what caused the bug. This check is needed because this function is called null every render until the element is rendered to the dom. 
Since I keep the element in state I also need to check if it has been set before, otherwise it will cause a recursive loop.